src/pages/index.js

```diff
@@ -54,31 +91,216 @@ variables.profileEditButton.addEventListener("click", () => {
+/* -------------------------------------------------------------------------------------------- */
+
+function createNewCard(card) {
+  variables.createCardButton.textContent = "Creating...";
+  addModalFormValidator.disableSubmitButton();
+  api
+    .createCard(card)
+    .then((cardData) => {
+      cardSection.addItem(createCard(cardData));
+      addCardPopup.close();
+      variables.addModalForm.reset();
+    })
+    .catch((error) => {
+      addModalFormValidator.enableSubmitButton();
+      console.error("Failed to create new card", error);
+    })
+    .finally(() => {
+      variables.createCardButton.textContent = "Create";
+    });
+}
+
+/* -------------------------------------------------------------------------- */
+/*               Call API to submit card info to be created                   */
+/* -------------------------------------------------------------------------- */
+
+const addCardPopup = new PopupWithForm(
   "#profile-add-modal",
   ({ title: name, link }) => {
-    renderCard({ name, link });
-    variables.addModalForm.reset();
+    createNewCard({ name, link });
     addModalFormValidator.disableSubmitButton();
   }
 );
 
-addFormPopup.setEventListeners();
+/* -------------------------------------------------------------------------- */
+/*                    Add Card Form Event Listener                            */
+/* -------------------------------------------------------------------------- */
+
+addCardPopup.setEventListeners();
 variables.addButton.addEventListener("click", () => {
-  addFormPopup.open();
+  addCardPopup.open();
 });
-//
-const cardSection = new Section(
-  {
-    items: data,
-    renderer: renderCard,
-  },
-  "#el-card-list"
+
+/* -------------------------------------------------------------------------- */
+/*                  Pass form inputs to validate *Avatar* form data           */
+/* -------------------------------------------------------------------------- */
+
+const avatarFormValidator = new FormValidator(
+  formConfig,
+  variables.avatarModalForm
+);
+avatarFormValidator.enableValidation();
+
+/* -------------------------------------------------------------------------- */
+/*                         Call API and pass info to update the avatar        */
+/* -------------------------------------------------------------------------- */
+
+const avatarPopupForm = new PopupWithForm(
+  "#edit-avatar",
+  ({ link: avatar }) => {
+    variables.avatarModalSaveButton.textContent = "Saving...";
+    api
+      .updateUserAvatar({ avatar })
+      .then((res) => {
+        user.updateAvatar(res.avatar);
+        avatarPopupForm.close();
+        variables.avatarModalForm.reset();
+      })
+      .catch((error) => {
+        console.error("Failed to update user avatar:", error);
+      })
+      .finally(() => {
+        variables.avatarModalSaveButton.textContent = "Save";
+        avatarFormValidator.disableSubmitButton();
```

Comment:
`avatarFormValidator.disableSubmitButton();` should be in `then` otherwise in case of an error you'll not be able to click the submit button for the 2nd time